### PR TITLE
feat(nats): nats plugin creation

### PR DESCRIPTION
**Pull request by tupe12334**
This pull request adds a NATS plugin with essential features like creating the services, module, and controllers of the NATS microservice

### CODE REVIEW

File: plugins/nestjs-nats/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  // }
+
+  // getBootstrapFunction(
+  //   astMainFile: namedTypes.File
+  // ): namedTypes.FunctionDeclaration {
+  //   const mainFileFunctionDeclarations = astMainFile.program.body.filter(
+  //     (node) => node.type === "FunctionDeclaration"
+  //   );
+  //   throw new Error("TODO: implement getBootstrapFunction");
+  // }
```
Comment (abrl91) [inline plugins/nestjs-nats/src/index.ts:110]: why this is in a comment? if you don't use it delete it please

File: plugins/nestjs-nats/src/static/nats.module.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Module } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { ClientsModule } from "@nestjs/microservices";
+import { NatsController } from "./nats.controller";
+import { natsClientModuleFactory } from "../templates/nats.module.factory";
```
Comment (mulygottlieb) [inline plugins/nestjs-nats/src/static/nats.module.ts:5]: Fix import to work on the generated code tree structure

File: plugins/nestjs-nats/src/templates/nats.module.factory.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { ConfigService } from "@nestjs/config";
+import { ClientProvider, Transport } from "@nestjs/microservices";
+
+declare const SERVICE_NAME: string;
+
+export const natsClientModuleFactory = (
+  configService: ConfigService
+): ClientProvider => {
+  const natsServersString = configService.get<string>("NASTS_SERVERS");
```
Comment (mulygottlieb) [inline plugins/nestjs-nats/src/templates/nats.module.factory.ts:9]: NASTS -> NATS
SERVERS -> SERVER

**Review by mulygottlieb — APPROVED** [review]

File: plugins/nestjs-nats/package.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
     "license": "Apache-2.0",
     "devDependencies": {
         "@amplication/code-gen-types": "^1.1.19",
-        "@amplication/code-gen-utils": "^0.0.5-beta.0",
```
Comment (mulygottlieb) [inline plugins/nestjs-nats/package.json]: Did you intentionally take it out of beta?

Comment (tupe12334) [reply plugins/nestjs-nats/package.json]: Please pull the changes to your local clone. We use the `^0.0.5` I changed that after we publish the version

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**